flink-streaming-java/src/main/java/org/apache/flink/streaming/util/typeutils/FieldAccessorFactory.java

```diff
@@ -38,6 +43,12 @@
 @Internal
 public class FieldAccessorFactory implements Serializable {
```

Comment:
Have you thought about splitting this class into a new method in `TypeInformation`? This way we would not need to perform class lookup but can simply let the corresponding TypeInformation in Scala code decide. In any case, such a method can be marked immediatly as deprecated because we want to get rid of these field accessors anyway.

Reply:
I haven't, no. Do you mean some form of `TypeInformation#getFieldAccessor`?
If the removal wasn't planned then I would definitely go for that, but as is it seems like the approach in the PR is less work?

Reply:
Yes, some form of `TypeInformation#getFieldAccessor` could be less error-prone instead of classpath wide look ups. But I don't have a strong opinion here as this would be one of the first things we should remove in a Flink 2.0 anyway.

Reply:
That's fair, the approach in the PR leaves some room for mistakes.
But polluting a Public interface with an immediately deprecated method for a single use-case is a bit icky isn't it.

Reply:
Am I understanding things correctly that you're overall fine with the PR and I could go ahead with merging it?